### PR TITLE
ML Rule Suppression UI Improvements

### PR DESCRIPTION
**Pull request by rylnd**
## Summary

1. Re-enables and adds additional ML cypress tests

1. Adds ML fields to Define Step
    Screenshot 2024-06-06 at 5 14 17 PM

1. Disables suppression UI when no relevant ML jobs are enabled
    Screenshot 2024-06-17 at 11 26 01 PM

1. Adds warning text when _some_ relevant ML jobs are not enabled
    Screenshot 2024-06-17 at 11 26 16 PM

### CODE REVIEW

File: x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/machine_learning/machine_learning_setup.ts [hunk 1]
```diff
@@ -22,7 +23,7 @@ export const executeSetupModuleRequest = async ({
     .set(getCommonRequestHeader('1'))
     .send({
       prefix: '',
-      groups: ['auditbeat'],
+      groups: [ML_GROUP_ID],
```
Comment (yctercero) [inline x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/machine_learning/machine_learning_setup.ts:26]: Amazing job sticking with it to figure this out. Woohoo to finally having our ML tests able to run.

File: x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/translations.tsx [hunk 2]
```diff
@@ -234,6 +234,21 @@ export const EQL_SEQUENCE_SUPPRESSION_GROUPBY_VALIDATION_TEXT = i18n.translate(
   }
 );
 
+export const MACHINE_LEARNING_SUPPRESSION_DISABLED_LABEL = i18n.translate(
+  'xpack.securitySolution.detectionEngine.createRule.stepDefineRule.machineLearningSuppressionDisabledLabel',
+  {
+    defaultMessage: 'Machine Learning jobs must be running to enable alert suppression.',
```
Comment (nastasha-solomon) [inline x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/translations.tsx]: While this is technically true, I think we can still be more direct and tell users that they must act to enable suppression. 

```suggestion
    defaultMessage: 'To enable alert suppression, start one or more Machine Learning jobs.',
```

Another, more general option: 
To enable alert suppression, start Machine Learning jobs.

File: x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/translations.tsx [hunk 3]
```diff
@@ -234,6 +234,21 @@ export const EQL_SEQUENCE_SUPPRESSION_GROUPBY_VALIDATION_TEXT = i18n.translate(
...
+  }
+);
+
+export const MACHINE_LEARNING_SUPPRESSION_INCOMPLETE_LABEL = i18n.translate(
+  'xpack.securitySolution.detectionEngine.createRule.stepDefineRule.machineLearningSuppressionIncompleteLabel',
+  {
+    defaultMessage:
+      'This list of fields may be incomplete as some Machine Learning jobs are not running.',
```
Comment (nastasha-solomon) [inline x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/translations.tsx]: What should users do if they notice that certain fields are missing? We should prob include that suggestion so they have a path forward from here.

Comment (rylnd) [reply x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/translations.tsx]: We could provide the list of non-running ML jobs, but that could get verbose. As with the above, the solution is to start ML jobs.

Comment (nastasha-solomon) [reply x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/components/step_define_rule/translations.tsx]: Makes sense. I also wonder how useful that list would be for users who don't know which jobs to enable. 

How's this:

```suggestion
      'This list of fields might be incomplete as some Machine Learning jobs are not running. Start more jobs to add more fields.',
```

**Pull request merged**